ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -25,12 +25,15 @@
   private final StoreKey key;
   private final long size;
   private final long expirationTimeInMs;
+  //TODO Replace booleans with enum defining MessageInfoType {PUT, DELETE, TTL_UPDATE, UNDELETE}
   private final boolean isDeleted;
   private final boolean isTtlUpdated;
+  private final boolean isUndeleted;
   private final Long crc;
   private final short accountId;
   private final short containerId;
   private final long operationTimeMs;
+  private final short lifeVersion;
```

Comment:
@zzmao could you ensure the term `lifeVersion` is reflected in your design doc? Let's make the terminology consistent everywhere. Thanks.

Reply:
ok. I will update.